.github/workflows/domain_ci.yml

```diff
@@ -145,3 +145,65 @@ jobs:
         run: pytest --no-header -v test/test_audio_examples.py
         env:
           PYTORCH_TEST_WITH_SLOW: 1
+
+  torcharrow:
+    if: ${{ github.repository_owner == 'pytorch' }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          # - macos-latest
+          - ubuntu-latest
+          # - windows-latest
+        python-version:
+          - 3.7
+          - 3.8
+          - 3.9
+    steps:
+      - name: Check if the CPU is supported
+        run: |
+          if [ "$RUNNER_OS" == "Linux" ]; then
+            cpu=$(lscpu | grep E5-2673)
+          elif [ "$RUNNER_OS" == "macOS" ]; then
+            cpu=$(sysctl -a | grep E5-2673)
+          else
+              echo "$RUNNER_OS is not supported"
+              exit 1
+          fi
+
+          if [ $cpu ]; then
+            echo "E5-2673 is not supported"
+            exit 1
```

Comment:
Not sure why the job doesn't exit
![image](https://user-images.githubusercontent.com/68879799/166249521-3cb7c6a6-2b3f-4e9e-91fd-67ad548e6af8.png)

Could you first use double bracket since we are using `bash` here? And, maybe try to do `if [[ -z $cpu ]]; then` to see if this condition is passed and the job exits normally.